core/dbt/parser/schemas.py

```diff
@@ -351,6 +352,23 @@ def create_test_node(
         column_name: Optional[str],
     ) -> ParsedSchemaTestNode:
 
+        HASH_LENGTH = 10
+
+        def get_hashable_md(
```

Comment:
I was hoping there would be a way to do this hash in a type-agnostic way instead, but I couldn't figure one out. Womp womp.

Reply:
On another note it would be nice to have a comment on this method.

Reply:
Yeah, there's not a lot of great options that are performant.  I'll add a comment to make clear what's happening.

Reply:
Added and updated.